### DISCUSSION TITLE
Does the project use deprecated api endpoints for the fcm cloud messaging platform?

### DISCUSSION THREAD

**Discussion by TeoMastro** [kreait/firebase-php#970] [Q&A] [answered]
### Describe the feature you would like to see

First of all sorry for the long sheet that you will see bellow, I just need a place to discuss some of the concerns I have.

I want to send push notifications from my php backend to my hybrid ionic mobile applications (I use a REST API so that the mobile and web app can communicate). I want to be able to use topics and all the features from my web server environment.
I did a lot of research on using the fcm cloud messaging from the backend of my php yii2 application and I could not manage to find an original package to do so. From what I saw in the [docs](https://firebase.google.com/docs/cloud-messaging/manage-topics), there are official libraries only for node, java, python, go and c#. That's how I ended up in this plugin.

My concern is that just by using api requests I can just access this endpoind => https://fcm.googleapis.com/v1/projects/project-id/messages:send, which just sends a notification to a single device or topic.
I want to be able to do topic management in the server.
From what I saw this plugin allows that. But does it use endpoints internally that are deprecated at the moment ([original migration guide](https://firebase.google.com/docs/cloud-messaging/migrate-v1))? Or maybe I am missing something???

For example:
---------------------------------------------------------------------------------------------------------------
use Kreait\Firebase\Messaging\CloudMessage;

$message = CloudMessage::new(); // Any instance of Kreait\Messaging\Message
$deviceTokens = ['...', '...' /* ... */];

/** @var Kreait\Firebase\Messaging\MulticastSendReport $sendReport **/
$sendReport = $messaging->sendMulticast($message, $deviceTokens);

Does this example from the doc use any deprecated api endpoint under the hood?

---------------------------------------------------------------------------------------------------------------

Thanks for your time and excuse me for the long text, I just could not exactly understand what is deprecated and what isn't in the FCM Messaging ecosystem. I think Google kinda mixes everything up and the original documentation is not very clear.

**Comment by jeromegamez** [comment, marked as answer]
Hey there, and thank you for taking time to make the text more than just a two-liner 🙏🏻 

In the 7.x releases, no deprecated/shut-down API endpoints are used. Previous releases continued to work until Firebase shut down the legacy API, so, if you want to use FCM, you need to use a 7.c release (preferably the latest one 😅)

As for the `sendMulticast()` method - it creates copies of the given message, modifies it to use the given registration tokens, and then forwards it to the `sendAll()` method.

The PHP SDK supports topic management as well.

The only drawback here is that the PHP SDK is unofficial and doesn't have _all_ the features the official SDKs have - this is due to the fact that I'm a one-man-team and my work is practically unpaid (I currently get $55 a month).

I hope this answers your question!

**Reply by TeoMastro** [reply]
Thank you very much for the reply and the awesome plugin!
